webapp/src/components/dashboard/products/UploadProduct.tsx

```diff
@@ -131,137 +130,144 @@ export default function UploadImage(props: UploadProductProps): JSX.Element {
     };
   }
 
-  return (
-    <React.Fragment>
-      <Container component="main" maxWidth="md" sx={{ mb: 4 }}>
-        <Paper
-          variant="outlined"
-          sx={{ my: { xs: 3, md: 6 }, p: { xs: 2, md: 3 } }}
-        >
-          <h1 style={{ margin: 8 }}>Upload a product</h1>
-
-          <Stack
-            direction="row"
-            spacing={2}
-            justifyContent="space-evenly"
-            alignItems="stretch"
+  if (
```

Comment:
Same here, you can just render UploadProduct and in a parent something like:

```suggestion
   if (localStorage.getItem("user.email") === null ||
        (localStorage.getItem("user.role") !== "admin" &&
         localStorage.getItem("user.role") !== "manager")
      return <Navigate to="/" />;
  else
      { props.children }
  )
```
Where props.children is this component: UploadProduct